backend/routes/userSession.js

```diff
@@ -0,0 +1,63 @@
+const express = require("express");
+const session = require('express-session');
+const {PrismaSessionStore} = require('@quixo3/prisma-session-store')
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+const userSessionRouter = express.Router();
+
+const ONE_DAY = 86400000;
+const userSession = (session({
+    secret: process.env.SESSION_SECRET,
+    saveUninitialized: false,
+    resave: false,
+    cookie: {
+        secure: false,
+        httpOnly: true,
+        maxAge: ONE_DAY
+    },
+    store: new PrismaSessionStore(
+        prisma,
+        {
+            checkPeriod: 2 * 60 * 1000,  //ms
+            dbRecordIdIsSessionId: true,
+            dbRecordIdFunction: undefined,
+        },
+    )
+}));
+
+const authenticateSession = ( async (req, res, next) => {
+    if (!req.session || !req.session.user) {
+        return res.status(401).json({message: "not authenticated"})
+    }
+    if (req.session.cookie.expires && req.session.cookie.expires < Date.now()) {
+        req.session.destroy()
+        return res.status(401).json({message: 'session expired'})
+    }
+
+    const userID = req.session.user;
+    try {
+        const session = await prisma.session.findUnique({
+            where: {id: userID}
+        });
```

Comment:
we shouldn't be doing it like this! we should be getting a session ID from the cookie right, we need to check that the cookie's session id exist, not find a session that is tied to a user.

conceptually you'll use `sid` to find session, then check that the session data's user exists and attach it to the request. 

In actuality, express-session should take care of it finding the session from the database already and have it attached as `req.session` so you shouldn't actually have to find your session manually.

i.e. this find session block is not needed iirc

Reply:
i worked out the stuff with checking for a cookie, im confused about where you said, "then check that the session data's user exists and attach it to the request.". i am attaching the user id to the request on the login, am i doing it wrong then or is it a misunderstanding